duckduckgo/Android
# Remote messaging - download and parse config (1)

Task/Issue URL: https://app.asana.com/0/72649045549333/1199214002677571/f

### Description

This PR is one of the parts of the Remote Messaging implementation.

In here, the reviewer will find:
- Remote messaging modules created
- Logic to schedule Remote Config worker to fetch remote config
- Parse remote Json config  as Data models
- Map Data Models into Domain models (includes some business logic rules to discard messages if mandatory information is missing)
- Creates a new DB to store Remote Config info (as version, last evaluation, etc.)

### Steps to test this PR

This PR does not include any UI component or user interactions.

- Ensure tests are green

_Ensure workers downloads config_
- Fresh install
- Open the app
- Ensure we have downloaded the remote config (checking logs you should see "RMF: Scheduling remote config worker")

### UI changes
No UI changes
